packages/expo/src/singleton.ts

```diff
@@ -14,7 +14,14 @@ type BuildClerkOptions = {
 };
 
 export function buildClerk({ key, tokenCache }: BuildClerkOptions): HeadlessBrowserClerk {
-  if (!clerk) {
+  // Support "hot-swapping" the Clerk instance at runtime. See JS-598 for additional details.
+  const hasKeyChanged = clerk && key !== clerk.publishableKey;
+
+  if (!clerk || hasKeyChanged) {
+    if (hasKeyChanged) {
+      tokenCache.clearToken?.(KEY);
+    }
```

Comment:
@nikosdouvlis Since the singleton is being initialized in line 28, doesn't that mean that wherever the singleton is imported and used, the reference will be updated with the new instance?

Reply:
Yeah, we are recreating the clerk instance below. The token cache change here was more of an assurance that we wipe out any existing state